### ISSUE TITLE
programs.atop: evaluation error "attribute 'mkSystemd' missing"

### ISSUE THREAD

**Issue by pschyska** [NixOS/nixpkgs#312196]
### Describe the bug
Since [49f6869](https://github.com/NixOS/nixpkgs/commit/49f6869f71fb2724674ccc18670bbde70843d43f), nixos/modules/programs/atop.nix has a bug: 
```
       error: attribute 'mkSystemd' missing
       at /nix/store/b1qqk0vlygqjq3k0vwfmwpz7v6n2xywb-source/nixos/modules/programs/atop.nix:123:23:
          122|           };
          123|           mkService = lib.mkSystemd "services";
             |                       ^
          124|           mkTimer = lib.mkSystemd "timers";
```

### Steps To Reproduce
1. eval nixos/modules/programs/atop.nix 

### Additional context
The reason is that `mkSystemd` wasn't coming from `with lib`, but defined in a let binding above.

### Notify maintainers

@acid-bong 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.9, NixOS, 24.05 (Uakari), genesis`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.22.0`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by acid-bong** [comment]
I apologize for the mistake. Fixed in #312181

**Comment by pschyska** [comment]
Just saw the other issue, thanks and no worries!
